integration-tester/tester/logic.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		return err
+	}
+
+	bsk, err := t.getValidBatchSigningKey(ownManifest)
+	if err != nil {
+		return err
+	}
+
+	phaPacketEncryptionKey, err := t.getValidPacketDecryptionKey(phaManifest)
+	if err != nil {
+		return err
+	}
+
+	facilPacketEncryptionKey, err := t.getValidPacketDecryptionKey(facilManifest)
+	if err != nil {
+		return err
+	}
+
+	err = t.purgeOldJobs()
+	if err != nil {
+		// Don't stop executing because of this error
+		log.Printf("Error when purging old successful jobs: %v\n", err)
+	}
+
+	job := t.createJob(phaManifest, facilManifest, phaPacketEncryptionKey, facilPacketEncryptionKey, bsk)
+
+	log.Println("Scheduling job...")
+	scheduledJob, err := t.kubeClient.ScheduleJob(job)
+	if err != nil {
+		return fmt.Errorf("scheduling job failed: %v", err)
+	}
+	if scheduledJob != nil {
+		log.Printf("\tscheduled job: %s\n", scheduledJob.Name)
+	}
+
+	return err
+}
+
+// purgeOldJobs will remove all successful jobs from the kubernetes namespace
+func (t *Tester) purgeOldJobs() error {
+	labelSelector := fmt.Sprintf("%s=%s,ingestor=%s", LabelKey, LabelValue, t.name)
+	fieldSelector := "status.successful=1"
+
+	err := t.kubeClient.RemoveJobCollection(metav1.DeleteOptions{}, metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	})
+
+	if err != nil {
+		return fmt.Errorf("purging old successful jobs failed: %v", err)
+	}
+
+	return nil
+}
+
+func (t *Tester) createJob(
+	phaManifest manifest.DataShareProcessorSpecificManifest,
+	facilManifest manifest.DataShareProcessorSpecificManifest,
+	phaPacketEncryptionPublicKey, facilPacketEncryptionPublicKey string,
+	bsk *corev1.Secret) *batchv1.Job {
+	trueP := true
+	env := []corev1.EnvVar{
+
+		{Name: "BATCH_SIGNING_PRIVATE_KEY",
+			ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: bsk.Name},
+				Key:                  "secret_key",
+			}}},
+		{Name: "BATCH_SIGNING_PRIVATE_KEY_IDENTIFIER", Value: bsk.Name},
+
+		{Name: "PHA_ECIES_PUBLIC_KEY", Value: phaPacketEncryptionPublicKey},
+		{Name: "FACILITATOR_ECIES_PUBLIC_KEY", Value: facilPacketEncryptionPublicKey},
+
+		{Name: "RUST_LOG", Value: "debug"},
+		{Name: "RUST_BACKTRACE", Value: "1"},
+		{Name: "AWS_ACCOUNT_ID", Value: t.awsAccountId},
+	}
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "integration-tester-facilitator",
+			Labels: map[string]string{
+				LabelKey:   LabelValue,
+				"ingestor": t.name,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:                "Never",
+					ServiceAccountName:           t.serviceAccountName,
+					AutomountServiceAccountToken: &trueP,
+					Containers: []corev1.Container{
+						{
+							Name:  "integration-tester",
+							Image: t.facilitatorImage,
+							Env:   env,
+							Args: []string{
```

Comment:
Let's just double check to make sure this makes sense

Reply:
Based on what's currently in `kubernetes.tf`, I think you have `own` and `peer` reversed. It's possible that `facilitator generate-ingestion-sample` is just wrong in its treatment of `own` and `peer` vs. facilitator and PHA.

I'm also not sure about the identity arguments, because I'm not sure where `facilManifest.IngestionIdentity` or `phaManifest.IngestionIdentity` come from, or what their values will be. But consider carefully whether they need values at all, and what their values should be based on whether the bucket being written to is a GCS or S3 bucket. The scheme for all these buckets and identities is documented [here](https://docs.google.com/document/d/1MdfM3QT63ISU70l63bwzTrxr93Z7Tv7EDjLfammzo6Q/edit#heading=h.m8nz1l5m6znk).